### PR TITLE
Metal3: Add validations to the Power Off modal

### PR DESCRIPTION
**Pull request by mareklibra**
The modal confirmation dialog for the BMH Power Off action newly shows validation
messages of selected potential issues which are preventing graceful power off.

### CODE REVIEW

**Comment by mareklibra** [comment]
01_validations

02_off

**Comment by mareklibra** [comment]
Cc @andybraren

File: frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx [hunk 1]
```diff
@@ -81,16 +89,83 @@ const isPowerOffSafe = (status: string) => {
...
+  if (status === NODE_STATUS_STARTING_MAINTENANCE) {
+    validations.push({
+      title: 'The node is starting maintenance.',
+      description:
+        'The node cannot be powered off gracefully until it is either in maintenance or healthy.',
+      level: 'info',
+    });
+  }
+
+  if (status === NODE_STATUS_STARTING_MAINTENANCE) {
```
Comment (andybraren) [inline frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx:126]: Should this be `NODE_STATUS_STOPPING_MAINTENANCE`? 🤷‍♂️

File: frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx [hunk 2]
```diff
@@ -81,16 +89,83 @@ const isPowerOffSafe = (status: string) => {
...
+    validations.push({
+      title: 'The node is stopping maintenance.',
+      description:
+        'The node cannot be powered off gracefully until it is either in maintenance or healthy.',
```
Comment (andybraren) [inline frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx:130]:
```suggestion
        'The node cannot be powered off gracefully while it is exiting maintenance.',
```

This is a weird scenario, but 🤷‍♂️

File: frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx [hunk 3]
```diff
@@ -81,16 +89,83 @@ const isPowerOffSafe = (status: string) => {
...
+
+  if (
+    nodePods?.data?.find((pod) =>
+      pod.metadata?.ownerReferences.find((or) => or.kind === DaemonSetModel.kind),
+    )
+  ) {
+    validations.push({
+      title: 'This node contains DaemonSet pods.',
+      description:
+        'These DaemonSet pods will prevent some pods from being moved. This should not prevent the host from powering off gracefully.',
```
Comment (andybraren) [inline frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx:143]:
```suggestion
        'These DaemonSets will prevent some pods from being moved. This should not prevent the host from powering off gracefully.',
```

File: frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx [hunk 4]
```diff
@@ -123,11 +200,12 @@ const PowerOffHostModal = withHandlePromise(
       return handlePromise(promise).then(close);
     };
 
-    const isUnderMaintenance = status === NODE_STATUS_UNDER_MAINTENANCE;
+    const isUnderMaintenance = status.status === NODE_STATUS_UNDER_MAINTENANCE;
     return (
-      <form onSubmit={submit} name="form" className="modal-content">
+      <form onSubmit={submit} name="form" className="modal-content metal3-poweroff-modal">
         <ModalTitle>Power Off Host {name}</ModalTitle>
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx:206]: done

File: frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx [hunk 5]
```diff
@@ -81,16 +89,83 @@ const isPowerOffSafe = (status: string) => {
...
+      description: 'The host cannot be powered off gracefully untils its health is restored.',
+      level: 'warning',
+    });
+  }
+
+  if (status === NODE_STATUS_STARTING_MAINTENANCE) {
+    validations.push({
+      title: 'The node is starting maintenance.',
+      description:
+        'The node cannot be powered off gracefully until it is either in maintenance or healthy.',
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx:121]: done

File: frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx (hunk 2 above)
Comment (mareklibra) [reply frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx:130]: Eagle eye :-) Thank you!

File: frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx (hunk 3 above)
Comment (mareklibra) [reply frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx:143]: done

**Comment by mareklibra** [comment]
> The "To power off gracefully..." sentence should appear above all validations.

done

> Is the "This node contains unmanaged static pods." warning validation possible? Hopefully it's very rare, but since it could be destructive it would be nice to have.

I propose to add this one in a follow up. I still have it on my mind but need to switch to another task for now so it would be unfortunate to block those existing ones on that. It will take me some time to play with the cluster. Wdyt?

**Comment by mareklibra** [comment]
@andybraren , I have updated the screenshots above

**Pull request merged**